routes/articles.js

```diff
@@ -22,74 +22,126 @@ router.get("/articles/new", (req, res) => {
+        article: results.toJSON(),
+        error: localError,
+        errorEdit: errorPUT,
+        errorDelete: errorDELETE
+      });
+    });
 });
 
-router.post("/articles", (req, res) => {
-  let isSuccessful = articlesDB.addArticle(req.body);
-  if (!isSuccessful) {
-    globalError = throwError(400, "Article cannot be added to database.".req);
-    errorPOST = true;
-    res.redirect("/articles/new");
-  } else {
-    res.redirect("/articles");
-  }
+router.post("/", (req, res) => {
+  let urlTitle = req.body.title.replace(/ /g, "-");
+  return req.db.Article.forge({
+    title: req.body.title,
+    body: req.body.body,
+    author: req.body.author,
+    urlTitle: urlTitle
+  })
+    .save(null, { method: "insert" })
+    .then(results => {
+      res.redirect("/articles");
+    })
+    .catch(err => {
+      globalError = throwError(
+        400,
+        "Article cannot be added to database.",
+        req
+      );
+      errorPOST = true;
+      res.redirect("/articles/new");
+    });
 });
 
-router.put("/articles/:title", (req, res) => {
-  let isSuccessful = articlesDB.editArticle(req.params.title, req.body);
-  if (!isSuccessful) {
-    globalError = throwError(400, "Article not found.", req);
-    errorPUT = true;
-    res.redirect(`/articles/${req.params.title}/edit`);
-  } else {
-    res.redirect(`/articles/${req.params.title}`);
+router.put("/:title", (req, res) => {
+  let urlTitle = req.body.title.replace(/ /g, "-");
+
+  if (urlTitle !== req.params.title) {
+    return req.db.Article.forge({
+      title: req.body.title,
+      body: req.body.body,
+      author: req.body.author,
+      urlTitle: urlTitle
+    })
+      .save(null, { method: "insert" })
+      .then(results => {
+        return req.db.Article.where({ urlTitle: req.params.title }).destroy();
+      })
+      .then(results => {
+        res.redirect(`/articles/${urlTitle}`);
+      })
+      .catch(err => {
+        globalError = throwError(400, "Article not found.", req);
+        errorPUT = true;
+        res.redirect(`${req.params.title}/edit`);
+      });
   }
-});
 
-router.delete("/articles/:title", (req, res) => {
-  let isSuccessful = articlesDB.deleteArticle(req.params.title);
-  if (!isSuccessful) {
-    globalError = throwError(
-      500,
-      `Cannot delete article called ${req.params.id}`,
-      req
-    );
-    errorDELETE = true;
-    res.redirect(`articles/${req.params.title}`);
-  } else {
-    success = true;
-    deletedArticle = isSuccessful;
+  return req.db.Article.where({ urlTitle: req.params.title })
+    .set({
+      title: req.body.title,
+      body: req.body.body,
+      author: req.body.author,
+      urlTitle: urlTitle
+    })
+    .save()
+    .then(results => {
+      res.redirect(`/articles/${urlTitle}`);
+    })
+    .catch(err => {
+      globalError = throwError(400, "Article not found.", req);
+      errorPUT = true;
+      res.redirect(`articles/${req.params.title}/edit`);
+    });
+});
```

Comment:
Sorry, can I get some clarification.  Are you referring to the fact that I call to the db twice in my if statement, or the fact that I have the if statement at all?

Reply:
So I think the best way to tackle this is, if the user changes the title, just remove the old entry and create a new one

Reply:
Also, try to make sure that the delete and the insert are in the same promise chain so that both actions aren't happening at the same time. But rather one after the other.